src/overlays/actors/ovl_En_Wood02/z_en_wood02.c

```diff
@@ -29,15 +60,392 @@ const ActorInit En_Wood02_InitVars = {
+            childPos.x = (sSpawnDistance[i] * sSpawnSin) + this->actor.initPosRot.pos.x;
+            childPos.y = this->actor.initPosRot.pos.y;
+            childPos.z = (sSpawnDistance[i] * sSpawnCos) + this->actor.initPosRot.pos.z;
+            if (func_80B3AF70(this, globalCtx, &childPos)) {
+                if ((this->unk_14E[i] & 0x80) != 0) {
+                    childParams = (0xFF00 | (this->actor.params + 1));
+                } else {
+                    childParams = (((this->drawType & 0xF0) << 4) | (this->actor.params + 1));
+                }
+                childWood = (EnWood02*)Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx,
+                                                          ACTOR_EN_WOOD02, childPos.x, childPos.y, childPos.z,
+                                                          this->actor.posRot.rot.x, *childSpawnAngle, 0, childParams);
+                if (childWood != NULL) {
+                    childWood->unk_14E[0] = i;
+                    this->unk_14E[i] |= 1;
+                    childWood->actor.projectedPos = this->actor.projectedPos;
+                } else {
+                    this->unk_14E[i] &= 0x80;
+                }
+            }
+        }
+    }
+}
+
+void EnWood02_Init(Actor* thisx, GlobalContext* globalCtx2) {
+
+    s16 spawnType;
+    f32 actorScale;
+    GlobalContext* globalCtx = globalCtx2;
+    EnWood02* this = THIS;
+    CollisionPoly* outPoly;
+    void* outBgId;
+    f32 floorY;
+    s16 extraRot;
+
+    spawnType = 0;
+    actorScale = 1.0f;
+    this->unk_14C = (this->actor.params >> 8) & 0xFF;
+
+    if (this->actor.initPosRot.rot.z != 0) {
+        this->actor.initPosRot.rot.z = (this->actor.initPosRot.rot.z << 8) | this->unk_14C;
+        this->unk_14C = -1;
+        this->actor.posRot.rot.z = this->actor.shape.rot.z = 0;
+    } else if (this->unk_14C & 0x80) {
+        this->unk_14C = -1;
+    }
+
+    this->actor.params &= 0xFF;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    if (this->actor.params < 0xB) { // Tree
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    }
+
+    switch (this->actor.params) {
+        case BUSH_GREEN_LARGE_SPAWNER:
+        case BUSH_BLACK_LARGE_SPAWNER:
+            spawnType = 1;
+        case BUSH_GREEN_LARGE_SPAWNED:
+        case BUSH_BLACK_LARGE_SPAWNED:
+            spawnType++;
+        case TREE_CONICAL_LARGE:
+        case BUSH_GREEN_LARGE:
+        case BUSH_BLACK_LARGE:
+            actorScale = 1.5f;
+            this->actor.uncullZoneForward = 4000.0f;
+            this->actor.uncullZoneScale = 2000.0f;
+            this->actor.uncullZoneDownward = 2400.0f;
+            break;
+        case TREE_CONICAL_SPAWNER:
+        case TREE_OVAL_YELLOW_SPAWNER:
+        case TREE_OVAL_GREEN_SPAWNER:
+        case BUSH_GREEN_SMALL_SPAWNER:
+        case BUSH_BLACK_SMALL_SPAWNER:
+            spawnType = 1;
```

Comment:
does spawn type correspond to type enum? if so you can use it

Reply:
That doesn't make sense here, since the spawner types start off with it being 1 instead of 0, and then it's incremented for both non-normal types. No idea why you'd do it that way, but there we are